File: Exemplos SmartPOS/App_IntentDigitalHUB_Java/Elgin_IntentDigitalHUB_SmartPOS_Java-font/app/src/main/java/com/elgin/java_intentdigitalhub_smartpos/IntentDigitalHubService/TERMICA/Commands/AbreConexaoImpressora.java
package com.elgin.java_intentdigitalhub_smartpos.IntentDigitalHubService.TERMICA.Commands;

import com.elgin.java_intentdigitalhub_smartpos.IntentDigitalHubService.TERMICA.TermicaCommand;
import com.google.gson.JsonObject;

public class AbreConexaoImpressora extends TermicaCommand {
    final private int tipo;
    final private String modelo;
    final private String conexao;
    final private int parametro;

    public AbreConexaoImpressora(int tipo, String modelo, String conexao, int parametro) {
        super("AbreConexaoImpressora");
        this.tipo = tipo;
        this.modelo = modelo;
        this.conexao = conexao;
        this.parametro = parametro;
    }

    @Override
    protected JsonObject functionParametersJson() {
        JsonObject functionParametersJson = new JsonObject();

        functionParametersJson.addProperty("tipo", this.tipo);
        functionParametersJson.addProperty("modelo", this.modelo);
        functionParametersJson.addProperty("conexao", this.conexao);
        functionParametersJson.addProperty("parametro", parametro);

        return functionParametersJson;
    }
}

File: Exemplos SmartPOS/App_IntentDigitalHUB_Kotlin/Elgin_IntentDigitalHUB_SmartPOS_Kotlin-font/app/src/main/java/com/elgin/kotlin_intent_digital_hub_smartpos/IntentDigitalHubService/TERMICA/Commands/AbreConexaoImpressora.kt
package com.elgin.kotlin_intent_digital_hub_smartpos.IntentDigitalHubService.TERMICA.Commands

import com.elgin.kotlin_intent_digital_hub_smartpos.IntentDigitalHubService.TERMICA.TermicaCommand
import com.google.gson.JsonObject

class AbreConexaoImpressora(
    private val tipo: Int,
    private val modelo: String,
    private val conexao: String,
    private val parametro: Int) :
    TermicaCommand("AbreConexaoImpressora") {
    override fun functionParametersJson(): JsonObject {
        val functionParametersJson = JsonObject()
        functionParametersJson.addProperty("tipo", tipo)
        functionParametersJson.addProperty("modelo", modelo)
        functionParametersJson.addProperty("conexao", conexao)
        functionParametersJson.addProperty("parametro", parametro)
        return functionParametersJson
    }
}